onnxruntime/core/mlas/lib/mlasi.h

```diff
@@ -97,6 +97,11 @@ Module Name:
 // Select the threading model.
 //
 
+#ifdef _OPENMP
```

Comment:
why is this needed?

Reply:
As mentioned in the patch description MLAS THREADPOOL broke OPENMP support. More specifically, when OPENMP is configured without this fix, ony one core is used at runtime.

Reply:
The default is to use the Eigen threadpool and the default size for that is half of std::hardware_concurrency (see CreateThreadPool). But as it is, CreateThreadPool wouldn't create a threadpool because your dual core system would compute thread_pool_size = 1. Normally, you could override this in the session options, but that doesn't help here. @snnn can you review CreateThreadPool behavior?

It shouldn't be that building with OPENMP automatically disables the session threadpool.

Reply:
"when OPENMP is configured without this fix, ony one core is used at runtime."  Only one core is used in OpenMP or the session thread pool?

---
In reply to: [339513366](https://github.com/microsoft/onnxruntime/pull/2258#discussion_r339513366) [](ancestors = 339513366)

Reply:
We use the C API example to validate the multicore capability:
https://github.com/microsoft/onnxruntime/blob/master/csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/C_Api_Sample.cpp

Before MLAS thread pool changes introduced in v0.4.0 the inference scaled on multiple cores (we are using USE_OPENMP=ON). We validated this both with 2xA72 and also with 4xA53 cores.
After the changes the inference is limited on a single core. We also tried to increase the number of thread from SetIntraOpNumThreads() without success.
https://github.com/microsoft/onnxruntime/blob/master/csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/C_Api_Sample.cpp#L35 

We could not get other hints from the documentation which is pretty thin. Is this a regression issue?

Reply:
@snnn any comments about the thread behavior @mikecaraman describes?

Reply:
Did you set openmp envs properly ?

Reply:
OMP_NUM_THREADS environment variable does not have an impact of the number of cores used on our platform but the issue seems to be fixed after v0.5.0, so all thinks are good without the patch.

Reply:
Yes, I already revert it locally.

Reply:
re-opening for tracking purposes. will close when the changes are pushed.

Reply:
Changes pushed.